pubnub-gson/src/integrationTest/java/com/pubnub/api/integration/RetryConfigurationIntegrationTest.java

```diff
@@ -0,0 +1,63 @@
+package com.pubnub.api.integration;
+
+import com.pubnub.api.PNConfiguration;
+import com.pubnub.api.PubNub;
+import com.pubnub.api.PubNubException;
+import com.pubnub.api.enums.PNStatusCategory;
+import com.pubnub.api.integration.util.BaseIntegrationTest;
+import com.pubnub.api.models.consumer.PNStatus;
+import com.pubnub.api.retry.RetryConfiguration;
+import com.pubnub.api.retry.RetryableEndpointGroup;
+import com.pubnub.api.v2.callbacks.StatusListener;
+import org.jetbrains.annotations.NotNull;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.pubnub.api.integration.util.Utils.randomChannel;
+import static org.junit.Assert.assertTrue;
+
+public class RetryConfigurationIntegrationTest extends BaseIntegrationTest {
+    @Test
+    public void whenRetryConfigurationIsDefinedShouldGetProperStatus() throws InterruptedException, PubNubException {
+        AtomicBoolean success = new AtomicBoolean();
+        PNConfiguration pnConfiguration = getBasicPnConfiguration();
+        pnConfiguration.setOrigin("ps.pndsn_notExisting_URI.com"); // we want to trigger UnknownHostException to initiate retry
+        pnConfiguration.setRetryConfiguration(new RetryConfiguration.Linear(1,2 ));
+        pnConfiguration.setHeartbeatInterval(1);
+        pubNub = getPubNub(pnConfiguration);
+
+        pubNub.addListener(new StatusListener() {
+            @Override
+            public void status(@NotNull PubNub pubnub, @NotNull PNStatus pnStatus) {
+                assertTrue(pnStatus.getCategory() == PNStatusCategory.PNConnectionError);
+                success.set(true);
+            }
+        });
+
+        pubNub.subscribe().channels(Arrays.asList(randomChannel())).execute();
+        Thread.sleep(10000);
+    }
+
+    @Test
+    public void canCreateRetryConfigurationsWithDifferentParameters() {
+        int delayInSec = 2;
+        int minDelay = 3;
+        int maxDelay = 5;
+        int maxRetryNumber = 2;
+        List<RetryableEndpointGroup> excludedOperations = new ArrayList<>();
+
+        RetryConfiguration linearRetryConfigurationWithNoParams = new RetryConfiguration.Linear();
```

Comment:
I found other strange thing. I am able to do:
RetryConfiguration linearRetryConfigurationWithNoParams = new RetryConfiguration.Linear(1L, 1, excludedOperations, true);
without compilation error and this constructor in defined as internal in different module.

Please, could you check if it behaves the same in you IDE?

Reply:
Java doesn't know about Kotlin internal, it treats it as public. I'll see if I can deal with this in another way

Reply:
I have this fixed locally, I'll make the change after we merge your PR
If you're wondering the trick is to make this constructor private, and create a factory method for tests in a companion object that is marked `@JvmSynthetic`. If only Kotlin supported synthetic constructors 🙄